lib/models/custom_theme_mode.dart

```diff
@@ -1,12 +1,29 @@
 import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
 
 class CustomThemeMode extends ChangeNotifier {
-  ThemeMode _themeMode = ThemeMode.system;
+  final themePreference = "theme_preference";
+
+  ThemeMode _themeMode;
+
+  CustomThemeMode() {
+    _loadTheme();
+  }
 
   ThemeMode get getThemeMode => _themeMode;
 
-  void setThemeMode(ThemeMode data) {
+  Future<void> _loadTheme() async {
+    SharedPreferences.getInstance().then((prefs) {
+      int preferredThemeIndex = prefs.getInt(themePreference) ?? 0;
+      _themeMode = ThemeMode.values[preferredThemeIndex];
+      notifyListeners();
+    });
+  }
+
+  Future<void> setThemeMode(ThemeMode data) async {
     _themeMode = data;
+    var prefs = await SharedPreferences.getInstance();
+    prefs.setInt(themePreference, _themeMode.index);
     notifyListeners();
   }
```

Comment:
Suggestion; 

We can use [hive](https://pub.dev/packages/hive) for saving theme mode. Because it's faster and secure than shared preferences.

Reply:
[Easy Localization package](https://pub.dev/packages/easy_localization) uses shared preferences to store language and @mukireus already used shared preferences to store on boarding page status that's why i used it to store theme mode and notification preferences instead of hive, but yeah using hive will be better for the app's performance.